core/app/models/spree/credit_card.rb

```diff
@@ -46,12 +46,15 @@ def self.default
 
     def default
       Spree::Deprecation.warn("CreditCard.default is deprecated. Please use user.wallet.default_wallet_payment_source instead.", caller)
-      user.wallet.default_wallet_payment_source.payment_source == self
+      return false if user.nil?
+      user.wallet.default_wallet_payment_source.try!(:payment_source) == self
     end
 
     def default=(set_as_default)
       Spree::Deprecation.warn("CreditCard.default= is deprecated. Please use user.wallet.default_wallet_payment_source= instead.", caller)
-      if set_as_default # setting this card as default
+      if user.nil?
```

Comment:
Also the [AddPaymentSourcesToWallet](https://github.com/solidusio/solidus/blob/ec58304541ad1655818042be0ea2ecb5863c7211/core/app/models/spree/wallet/add_payment_sources_to_wallet.rb#L14) code is a no-op if `order.user` is nil.